### PR TITLE
Fix: actor getting 2 queues for the same role

### PR DESCRIPTION
**Pull request by MadsDue**
Our pipeline behaviours creates the actor queue based on the outgoing messages by checking if none already exists for the current actor + role through a raw sql query. 

While new actor queue are not yet flushed to the db. 

Case: 
2 out going message: 

1. iteration: 
- No actor queue in db
- Creates a new actor queue in EF cache

2. iteraction
- No actor queue in db
- Creates a new actor queue in EF cache

UnitOfWorkBehaviour => stores both queues in db.

### CODE REVIEW

File: source/ApplyDBMigrationsApp/ApplyDBMigrationsApp.csproj [hunk 1]
```diff
@@ -115,6 +115,7 @@ limitations under the License.
     <EmbeddedResource Include="Scripts\Model\202308071047 Remove SettlementVersion from AggregatedMeasureDataProcesses.sql" />
     <EmbeddedResource Include="Scripts\Model\202308171048 Add ProcessDate index on ReceivedIntegrationEvents.sql" />
     <EmbeddedResource Include="Scripts\Model\202308171049 Add ProcessDate index on ReceivedInboxEvents.sql" />
+    <EmbeddedResource Include="Scripts\Model\202308221313 Add Constraint for ActorNumber and ActorRole on OueuedInternalCommandscessDate.sql" />
```
Comment (RasmusGraabaek) [inline source/ApplyDBMigrationsApp/ApplyDBMigrationsApp.csproj:118]:
```suggestion
    <EmbeddedResource Include="Scripts\Model\202308221313 Add Constraint for ActorNumber and ActorRole on ActorMessageQueues.sql" />
```

File: source/Infrastructure/Configuration/CompositionRoot.cs [hunk 2]
```diff
@@ -85,6 +85,7 @@ private CompositionRoot(IServiceCollection services)
             services.AddScoped<IUnitOfWork, UnitOfWork>();
             services.AddScoped<IOutgoingMessageRepository, OutgoingMessageRepository>();
             services.AddScoped<IFeatureFlagProvider, FeatureFlagProviderProvider>();
+            services.AddMemoryCache();
```
Comment (RasmusGraabaek) [inline source/Infrastructure/Configuration/CompositionRoot.cs:88]:
```suggestion
```

File: source/Infrastructure/OutgoingMessages/Queueing/ActorMessageQueueRepository.cs [hunk 3]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 using System;
+using System.Linq;
```
Comment (RasmusGraabaek) [inline source/Infrastructure/OutgoingMessages/Queueing/ActorMessageQueueRepository.cs:16]:
```suggestion
```

File: source/IntegrationTests/Application/Transactions/AggregatedMeasureData/WhenAnAcceptedResultIsAvailableTests.cs [hunk 4]
```diff
@@ -14,8 +14,12 @@
 
 using System;
 using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
+using Application.Configuration;
 using Application.Configuration.DataAccess;
+using Application.Configuration.TimeEvents;
+using Dapper;
```
Comment (RasmusGraabaek) [inline source/IntegrationTests/Application/Transactions/AggregatedMeasureData/WhenAnAcceptedResultIsAvailableTests.cs:22]:
```suggestion
using System.Threading.Tasks;
using Application.Configuration.DataAccess;
```

File: source/IntegrationTests/Application/Transactions/AggregatedMeasureData/WhenAnAcceptedResultIsAvailableTests.cs [hunk 5]
```diff
@@ -30,6 +34,8 @@
 using Infrastructure.OutgoingMessages.Common;
 using IntegrationTests.Assertions;
 using IntegrationTests.Fixtures;
+using MediatR;
+using Microsoft.Extensions.DependencyInjection;
```
Comment (RasmusGraabaek) [inline source/IntegrationTests/Application/Transactions/AggregatedMeasureData/WhenAnAcceptedResultIsAvailableTests.cs:38]:
```suggestion
```

File: source/IntegrationTests/Infrastructure/Configuration/InternalCommands/InternalCommandProcessorTests.cs [hunk 6]
```diff
@@ -71,21 +73,72 @@ public async Task When_execution_fails_the_exception_is_logged_and_command_is_ma
...
 
-    private void AssertIsProcessed(InternalCommand command)
+    [Fact]
+    public async Task When_parallels_internal_command_processor_handles_same_commands()
+    {
+        // Arrange
+        var serviceScopeFactory = GetService<IServiceScopeFactory>();
+        using var newScope = serviceScopeFactory.CreateScope();
+        var internalCommandProcessor = newScope.ServiceProvider.GetRequiredService<InternalCommandProcessor>();
+        var commandThatThrows = new TestCreateOutgoingMessageCommand(1);
```
Comment (RasmusGraabaek) [inline source/IntegrationTests/Infrastructure/Configuration/InternalCommands/InternalCommandProcessorTests.cs:86]: command

File: source/Infrastructure/Configuration/InternalCommands/InternalCommandProcessor.cs [hunk 7]
```diff
@@ -95,7 +119,7 @@ private async Task MarkAsFailedAsync(QueuedInternalCommand queuedCommand, string
                 "WHERE [Id] = @Id",
                 new
                 {
-                    NowDate = DateTime.UtcNow,
+                    NowDate = _systemDateTimeProvider.Now().ToDateTimeUtc(),
```
Comment (lasrinnil) [inline source/Infrastructure/Configuration/InternalCommands/InternalCommandProcessor.cs:122]: Why this change? What is the difference?, please enlighten me :D

File: source/Infrastructure/OutgoingMessages/Queueing/MessageEnqueuer.cs [hunk 8]
```diff
@@ -46,7 +43,5 @@ public async Task EnqueueAsync(OutgoingMessage message)
         }
 
         messageQueue.Enqueue(message);
-
-        _outgoingMessageRepository.Add(message);
```
Comment (lasrinnil) [inline source/Infrastructure/OutgoingMessages/Queueing/MessageEnqueuer.cs:50]: Erhm is this ok?

File: source/IntegrationTests/Application/OutgoingMessages/WhenEnqueueingTests.cs [hunk 9]
```diff
@@ -112,9 +118,8 @@ private static OutgoingMessage CreateOutgoingMessage()
 
     private async Task EnqueueMessage(OutgoingMessage message)
     {
-        var messageEnqueuer = GetService<MessageEnqueuer>();
-
-        await messageEnqueuer.EnqueueAsync(message);
+        await _messageEnqueuer.EnqueueAsync(message);
+        _outgoingMessageRepository.Add(message);
```
Comment (lasrinnil) [inline source/IntegrationTests/Application/OutgoingMessages/WhenEnqueueingTests.cs:122]: ? Perhaps you should give me a walkthrough of these changes to calling the _outgoingMessageRepository

**Review by lasrinnil — COMMENTED** [review]
Great job! I think we should go over it together - instead of me asking too many questions here :)
